core/server/api/utils.js

```diff
@@ -209,7 +209,13 @@ utils = {
             var unsafeAttrObject = unsafeAttrNames && _.has(options, 'data.[' + docName + '][0]') ? _.pick(options.data[docName][0], unsafeAttrNames) : {},
                 permsPromise = permissions.canThis(options.context)[method][singular](options.id, unsafeAttrObject);
 
-            return permsPromise.then(function permissionGranted() {
+            return permsPromise.then(function permissionGranted(result) {
+                // Allow the permissions function to return a list of disallowed fields.
```

Comment:
What's the difference between an `unsafeAttr` and a disallowed field?

Reply:
an unsafeAttr can trigger a NoPermissions exception in particular conditions, without excluding the field from the request, whereas a disallowed field is basically excluding that field from the request body altogether, without triggering a NoPermission exception. unsafeAttrs are also marked unsafe regardless of role and then it's up to the permissible function to determine whether or not to throw - disallowedFields can be different based on what the model permissible function determines.